src/wazuh_modules/syscollector/tests/sysCollectorFlatbuffers/syscollectorFb_test.cpp

```diff
@@ -0,0 +1,286 @@
+/*
+ * Wazuh SyscollectorImp
+ * Copyright (C) 2015, Wazuh Inc.
+ * August 1, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "syscollectorFb_test.h"
+
+constexpr auto SYNC_HMSG_TEST_PATH {"sync_hmsg.mom"};
+constexpr auto SYNC_PMSG_TEST_PATH {"sync_pmsg.mom"};
+
+constexpr auto COMPONENT {"syscollector"};
+constexpr auto TYPE {"state"};
+
+void SyscollectorFbTest::SetUp() {};
+
+void SyscollectorFbTest::TearDown() {};
+
+TEST(SyscollectorFbTest, createBinaryHotfix)
+{
+    flatbuffers::FlatBufferBuilder builder;
+
+    auto component = builder.CreateString(COMPONENT);
+    auto type = builder.CreateString(TYPE);
+
+    auto checksum = builder.CreateString("md5sha1md5sha1");
+    auto hotfix = builder.CreateString("KBXXXXX");
+    auto scan_time = builder.CreateString("00");
+
+    auto data = CreateHotfixAttribute(builder, checksum, hotfix, scan_time);
+
+    auto index = builder.CreateString("00");
+    auto timestamp = builder.CreateString("00/00/0000T00:00:00.000");
+
+    auto hotfixData = CreateHotfixData(builder, data, index, timestamp);
+
+
+    auto sync_msg = CreateSyncMsg(builder, component, Data_HotfixData, hotfixData.Union(), type);
+
+    builder.Finish(sync_msg);
+
+    uint8_t* buf = builder.GetBufferPointer();
+    int size = builder.GetSize();
+
+    std::ofstream ofile(SYNC_HMSG_TEST_PATH, std::ios::binary);
+    ofile.write((char*)buf, size);
+    ofile.close();
+}
+
+TEST(SyscollectorFbTest, readBinaryHotfix)
+{
+    std::ifstream infile;
+    infile.open(SYNC_HMSG_TEST_PATH, std::ios::binary | std::ios::in);
+    infile.seekg(0, std::ios::end);
+    int length = infile.tellg();
+    infile.seekg(0, std::ios::beg);
+    char* raw_data = new char[length];
+    infile.read(raw_data, length);
+    infile.close();
+
+    auto sync_msg = GetSyncMsg(raw_data);
+
+    EXPECT_STREQ(sync_msg->component()->c_str(), COMPONENT);
+    EXPECT_STREQ(sync_msg->type()->c_str(), TYPE);
+    EXPECT_EQ(sync_msg->data_type(), Data_HotfixData);
+
+    auto data = static_cast<const HotfixData*>(sync_msg->data());
+
+    EXPECT_STREQ(data->index()->c_str(), "00");
+    EXPECT_STREQ(data->timestamp()->c_str(), "00/00/0000T00:00:00.000");
+    EXPECT_STREQ(data->data()->checksum()->c_str(), "md5sha1md5sha1");
+    EXPECT_STREQ(data->data()->hotfix()->c_str(), "KBXXXXX");
+    EXPECT_STREQ(data->data()->scan_time()->c_str(), "00");
+
+    std::remove(SYNC_HMSG_TEST_PATH);
+    delete []raw_data;
+}
+
+TEST(SyscollectorFbTest, createBinaryPackageUnix)
```

Comment:
This test doesn't do any assertion. It only creates the file that is used in the test "readBinaryPackageUnix". So, this test could be merged with "readBinaryPackageUnix" in a unique test.

Reply:
Fix it, thanks !